server/planning/types/locations.py

```diff
@@ -0,0 +1,79 @@
+from datetime import datetime
+from typing import Any, Annotated
+from pydantic import Field
+
+from planning.types.base import BasePlanningModel
+
+from superdesk.utc import utcnow
+from superdesk.core.resources import fields, dataclass
+from superdesk.core.resources.validators import validate_data_relation_async, validate_iunique_value_async
+from superdesk.core.utils import generate_guid, GUID_NEWSML
+
+
+@dataclass
+class Position:
+    latitude: float | None = None
+    longitude: float | None = None
+    altitude: int | None = None
+    gps_datum: str | None = None
+
+
+@dataclass
+class Address:
+    title: str | None = None
+    line: list[str] | None = None
+    suburb: str | None = None
+    city: str | None = None
+    state: str | None = None
+    postal_code: str | None = None
+    country: str | None = None
+    locality: str | None = None  # Aka city
+    area: str | None = None
+    external: dict | None = None
+    boundingbox: list | None = None
+    address_type: str | None = Field(alias="type", default=None)
+
+
+class LocationResourceModel(BasePlanningModel):
+    guid: Annotated[fields.Keyword, Field(default_factory=lambda: generate_guid(type=GUID_NEWSML))]
```

Comment:
This field also the `validate_iunique_value_async` validation, to ensure data integrity that this field will always be unique, as we support this field being provided outside the model (aka not using the default factory)